.github/scripts/findUnusedStyles.ts

```diff
@@ -0,0 +1,561 @@
+                if (ts.isTemplateExpression(node)) {
+                    for (const span of node.templateSpans) {
+                        if (ts.isIdentifier(span.expression) && span.expression.text === key) {
+                            found = true;
+                            return;
+                        }
+                        // Also check for property access in template spans
+                        if (ts.isPropertyAccessExpression(span.expression) && ts.isIdentifier(span.expression.name) && span.expression.name.text === key) {
+                            found = true;
+                            return;
+                        }
+                    }
+                }
+
+                // Tagged template literals: css`${key}`
+                if (ts.isTaggedTemplateExpression(node) && ts.isTemplateExpression(node.template)) {
+                    for (const span of node.template.templateSpans) {
+                        if (ts.isIdentifier(span.expression) && span.expression.text === key) {
+                            found = true;
+                            return;
+                        }
+                    }
+                }
+
+                ts.forEachChild(node, visit);
+            };
+
+            visit(sourceFile);
+            return found;
+        } catch (error) {
+            // Fallback to simple text search if AST parsing fails
+            console.warn(`Warning: Could not parse ${file} for usage detection, falling back to text search`);
+            return content.includes(key);
+        }
+    }
+
+    private getUnusedStyles(): StyleDefinition[] {
+        // Whatever remains in styleDefinitions are the unused styles
+        const unused = Array.from(this.styleDefinitions.values());
+        return unused.sort((a, b) => a.file.localeCompare(b.file) || a.line - b.line);
+    }
+}
+
+// CLI interface
+function main() {
+    const rootDir = process.cwd();
+
+    try {
+        const finder = new ComprehensiveStylesFinder(rootDir);
+        const unusedStyles = finder.findUnusedStyles();
+
+        if (unusedStyles.length === 0) {
+            console.log('✅ No unused styles found!');
+            process.exit(0);
+        }
+
+        console.log(`Found ${unusedStyles.length} unused styles:`);
+        console.log('');
+
+        const groupedByFile: Record<string, StyleDefinition[]> = {};
+        for (const style of unusedStyles) {
+            if (!groupedByFile[style.file]) {
+                groupedByFile[style.file] = [];
+            }
+            groupedByFile[style.file].push(style);
+        }
+
+        for (const [file, styles] of Object.entries(groupedByFile)) {
+            console.log(`📁 ${file}:`);
+            for (const style of styles) {
+                console.log(`  - ${style.key} (line ${style.line}:${style.column})`);
+            }
+            console.log('');
+        }
+
+        if (process.argv.includes('--exit-code')) {
```

Comment:
NAB because you have only one flag and this is a very simple check, but I wrote `scripts/utils/CLI.ts` in the hopes that we could use it for arg parsing for all our TypeScript CLIs.

Also, I'm wondering if there's any case where we wouldn't want to exit w/ status code 1 if there are unused styles. That's how I'd expect this script to work, and I'm not sure if there's a use-case for how it behaves without the `--exit-code` flag